src/components/common/Extensions/UserHubButton/UserHubButton.tsx

```diff
@@ -56,6 +72,11 @@ const UserHubButton: FC<UserHubButtonProps> = ({
     );
 
   useDisableBodyScroll(visible && isMobile);
+  const handleButtonClick = () => {
```

Comment:
Is this the pattern we are using moving forward? I was hoping we could maybe be a bit smarter with the tracking. E.g. you could create a function, that wraps a function.

Here's an example:


```tsx
// This is also a bit annoying, maybe we can come up with a better, shorter abstraction?
// At least the advantage here is that this doesn't necessarily have to live inside the component (or this file, even)
const trackUserButtonClick = [
    AnalyticsEventType.CUSTOM_EVENT,
    AnalyticsEventCategory.USER,
    AnalyticsEventAction.CLICK,
    AnalyticsEventLabel.OPEN_USER_HUB,
];

// ...

const { track } = useAnalyticsContext();
// This will look much cleaner with only one function being passed in
return (
<div
  onClick={track(trackUserButtonClick, () => {
    setOpenItemIndex(undefined);
    toggleOff();
  })}
/>
)
```

Reply:
I honestly don't see a benefit in making a funcition call a callback after an event was tracked, it just makes it seem more complicated than it actually is.
I however agree on reworking the `trackEvent` API, I'll try to make it receive an object, and then we can reuse objects across the app if ever necessary